contrib/pkg/adm/managedns/enable.go

```diff
@@ -191,15 +206,59 @@ func (o *Options) Run(dynClient client.Client, args []string) error {
 	return nil
 }
 
-func podHasEnvVar(p corev1.Pod, envVar string) bool {
-	for _, ev := range p.Spec.Containers[0].Env {
-		if ev.Name == envVar {
-			return true
+func getUpdatedManagedDNSConfig(dynClient client.Client, previousGeneration int64) (string, error) {
+	deployment := &appsv1.Deployment{}
+	key := types.NamespacedName{Name: hiveAdmissionDeployment, Namespace: hiveNamespace}
+
+	log.Info("waiting for new hiveadmission deployment")
+	for i := 0; i < 24; i++ {
+		err := dynClient.Get(context.TODO(), key, deployment)
+		if err != nil {
+			log.WithError(err).Fatal("error getting hiveadmission deployment")
+			return "", err
```

Comment:
Is there a race condition where this may run prior to the operator creating the hiveadmission deployment for the first time? If that does happen, then the test will fail here since the `Get` will return a NotFound error.

Reply:
Yes! I'll change it to catch the IsNotFound case and keep waiting for a new deployment to appear.